case-server/src/main/java/com/powsybl/caseserver/CaseService.java

```diff
@@ -223,6 +226,31 @@ UUID importCase(MultipartFile mpf, boolean toPublic) {
         return caseUuid;
     }
 
+    UUID createCase(UUID sourceCaseUuid) {
+        try {
+            Path existingCaseFile = getCaseFile(sourceCaseUuid);
+            if (existingCaseFile == null || existingCaseFile.getParent() == null) {
+                throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Source case " + sourceCaseUuid + " not found");
+            }
+
+            UUID newCaseUuid = UUID.randomUUID();
+            Path newCaseUuidDirectory = existingCaseFile.getParent().getParent().resolve(newCaseUuid.toString());
+            Path newCaseFile;
+            Files.createDirectory(newCaseUuidDirectory);
+            newCaseFile = newCaseUuidDirectory.resolve(existingCaseFile.getFileName());
+            Files.copy(existingCaseFile, newCaseFile, StandardCopyOption.COPY_ATTRIBUTES);
+
+            CaseInfos existingCaseInfos = caseInfosService.getCaseInfosByUuid(sourceCaseUuid.toString()).orElseThrow();
+            CaseInfos caseInfos = createInfos(existingCaseInfos.getName(), newCaseUuid, existingCaseInfos.getFormat());
+            caseInfosService.addCaseInfos(caseInfos);
+            sendImportMessage(caseInfos.createMessage());
+            return newCaseUuid;
+
+        } catch (IOException e) {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN, "An error occured during the case file duplication");
```

Comment:
It's weird to throw a 403 for any IOException, forbidden should be only for access rights restriction and not any error.